core/src/main/java/com/amazon/opendistroforelasticsearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -123,19 +392,167 @@ private ExprValue exprDate(ExprValue exprValue) {
     }
   }
 
+  /**
+   * Name of the Weekday implementation for ExprValue.
+   * @param date ExprValue of Date type.
+   * @return ExprValue.
+   */
+  private ExprValue exprDayName(ExprValue date) {
+    if (date instanceof ExprStringValue) {
+      date = new ExprDateValue(date.stringValue());
+    }
+    return new ExprStringValue(
+        date.dateValue().getDayOfWeek().getDisplayName(TextStyle.FULL, Locale.getDefault()));
+  }
+
   /**
    * Day of Month implementation for ExprValue.
    * @param date ExprValue of Date type.
    * @return ExprValue.
    */
   private ExprValue exprDayOfMonth(ExprValue date) {
     if (date instanceof ExprStringValue) {
-      return new ExprIntegerValue(
-          new ExprDateValue(date.stringValue()).dateValue().getDayOfMonth());
+      date = new ExprDateValue(date.stringValue());
     }
     return new ExprIntegerValue(date.dateValue().getDayOfMonth());
   }
 
+  /**
+   * Day of Week implementation for ExprValue.
+   * @param date ExprValue of Date type.
+   * @return ExprValue.
+   */
+  private ExprValue exprDayOfWeek(ExprValue date) {
+    if (date instanceof ExprStringValue) {
+      date = new ExprDateValue(date.stringValue());
+    }
+    return new ExprIntegerValue(date.dateValue().getDayOfWeek().getValue() == 7 ? 1
```

Comment:
I think you can do without ternary.

`return new ExprIntegerValue((date.dateValue().getDayOfWeek().getValue() % 7) + 1)`

Reply:
done